### PR TITLE
Working commit for #3

### LINKED ISSUE #3: Add Linting and Lint code
@thekemkid

- Add linting, same as seneca
- Add Travis file
- Add tests via lab to cover all patterns

### PR DESCRIPTION
**Pull request by thekemkid**
- [x] Add linting, same as seneca
- [x] Add Travis file _completed by Isaac_
- [x] Add tests via lab to cover all patterns

Additional notes:
- [x] Add pre-commit hooks
- [x] Add coverage
- [x] Add coverage to travis

Resolves #3

### CODE REVIEW

**Comment by mcdonnelldean** [comment]
Added wip so I don't drive by accept as I'm known to do :D

File: github.js [hunk 1]
```diff
@@ -1,72 +1,64 @@
 /* Copyright (c) 2014-2015 Richard Rodger, MIT License */
```
Comment (mcdonnelldean) [inline github.js]: Looks epic, can we comment out this file ala, https://github.com/rjrodger/nodezoo-info/blob/code-change/info.js

File: github.js [hunk 2]
```diff
@@ -1,72 +1,64 @@
...
 })
 
-
-
-module.exports = function github( options ){
+module.exports = function github (options) {
   var seneca = this
 
   options = seneca.util.deepextend({
     token: 'YOUR_TOKEN_HERE'
```
Comment (mcdonnelldean) [inline github.js]: Change to '' so people pass it in the options and don't try to update this file

File: github.js [hunk 3]
```diff
@@ -76,48 +68,46 @@ module.exports = function github( options ){
...
       }
     )
   }
 
 
-  function cmd_parse( args, done ) {
-    var seneca  = this
+  function cmd_parse (args, done) {
+    // commented out unused var...
+    // var seneca = this
```
Comment (mcdonnelldean) [inline github.js]: remove

File: github.js [hunk 4]
```diff
@@ -76,48 +68,46 @@ module.exports = function github( options ){
...
 
     var m = /[\/:]([^\/:]+?)[\/:]([^\/]+?)(\.git)*$/.exec(args.giturl)
```
Comment (mcdonnelldean) [inline github.js]: m > match

File: github.js [hunk 5]
```diff
@@ -76,48 +68,46 @@ module.exports = function github( options ){
...
-    if( m ) {
-      return done( null, { user:m[1], repo:m[2] })
```
Comment (mcdonnelldean) [inline github.js]: can we get more info here? dump as much as you can

File: package.json [hunk 6]
```diff
@@ -15,12 +21,23 @@
   ],
```
Comment (mcdonnelldean) [inline package.json]: add yourself to contributors

File: srv/github-dev.js [hunk 7]
```diff
@@ -1,45 +1,47 @@
+'use strict'
 
-var HOST = process.env.HOST || 'localhost'
-var TOKEN = process.env.TOKEN || 'NO_TOKEN'
+// HOST is commented out because it is unusued
+// const HOST = process.env.HOST || 'localhost'
```
Comment (mcdonnelldean) [inline srv/github-dev.js]: remove

File: srv/github-dev.js [hunk 8]
```diff
@@ -1,45 +1,47 @@
...
+const TOKEN = process.env.TOKEN || 'NO_TOKEN'
```
Comment (mcdonnelldean) [inline srv/github-dev.js]: Comment out file

File: package.json (hunk 6 above)
Comment (thekemkid) [inline package.json]: resolved

**Comment by thekemkid** [comment]
again, updated. I didn't comment the srv file too much as I didn't really know what was going on there.

**Comment by mcdonnelldean** [comment]
Oh no! merge conflict!

**Comment by thekemkid** [comment]
@mcdonnelldean ping

**Pull request merged**